plugins/storage/volume/default/src/main/java/org/apache/cloudstack/storage/datastore/lifecycle/CloudStackPrimaryDataStoreLifeCycleImpl.java

```diff
@@ -250,15 +249,15 @@ public DataStore initialize(Map<String, Object> dsInfos) {
             parameters.setPath(hostPath.replaceFirst("/", ""));
             parameters.setUserInfo(userInfo);
         } else if (scheme.equalsIgnoreCase("PreSetup")) {
-            if (StringUtils.isNotBlank(hypervisorType) && HypervisorType.getType(hypervisorType).equals(HypervisorType.VMware)) {
+            if (hypervisorType.equals(HypervisorType.VMware)) {
```

Comment:
I do not know if hypervisorType is 100% not null.
maybe it is better to use `if (HypervisorType.VMware.equals(hypervisorType)) {` ?

Reply:
Wouldn't that hide improper calls to this function?

Reply:
no, @alexandru-bagu , it is common practice using equals to call it on the object guaranteed not to be null, in this case the enum member
```suggestion
            if (HypervisorType.VMware.equals(hypervisorType)) {
```

Reply:
You misunderstand. If someone calls this function without providing this parameter then they wouldn't know because there would be no error reported. Alas I will commit your suggestions.